hack/e2e.go

```diff
@@ -770,8 +785,28 @@ func finishRunning(stepName string, cmd *exec.Cmd) error {
 		log.Printf("Step '%s' finished in %s", stepName, time.Since(start))
 	}(time.Now())
 
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("error running %v: %v", stepName, err)
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("error starting %v: %v", stepName, err)
+	}
+
+	finished := make(chan error)
+
+	go func() {
+		finished <- cmd.Wait()
+	}()
+
+	for {
+		select {
+		case <-terminate.C:
+			terminate.Reset(time.Duration(0))  // Kill subsequent processes immediately.
+			cmd.Process.Kill()
+			return fmt.Errorf("Terminate testing after 15m after %s timeout during %s", *timeout, stepName)
+		case <-interrupt.C:
+			log.Printf("Interrupt testing after %s timeout. Will terminate in another 15m", *timeout)
+			terminate.Reset(time.Duration(15) * time.Minute)
```

Comment:
minor nit: I think usually this is just written `15 * time.Minute` - no need to `time.Duration`-wrap.